### PR TITLE
Feature/one version of libs

### LINKED ISSUE #81: What happens to existing aaf-shib-ext.jar if it's version number is bumped?
It must be _replaced_ to ensure no class loader issues.

### LINKED ISSUE #197: One version of libraries
The way Shibboleth build it's .war file and how it includes extra libraries it is possible to have multiple versions of these libraries loaded into the .war file. This situation needs to be avoided.

### PR DESCRIPTION
**Pull request by trsau**
Reorganized how external libraries are managed in the installer.

Tested with new install and upgrade.

close #197  
close #81

### CODE REVIEW

File: README_libraries.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+The Shibboleth Installer uses and deploys a number of libraries from external sources. This
+file documents these libraries.
+
+| Library | Current Version | Latest Version | Source |
+| ------- | --------------- | -------------- | ------ |
+| Connector-J | 5.1.35 | 5.1.41 | https://dev.mysql.com/downloads/connector/j/ |
+| jetty9-dta-ssl | 1.0.0 | 1.0.0 | https://build.shibboleth.net/nexus/content/repositories/releases/net/shibboleth/utilities/jetty9/jetty9-dta-ssl/ |
+| commons_dbcp2 | 2.1.1 | 2.1.1 | https://commons.apache.org/proper/commons-dbcp/download_dbcp.cgi |
+| commons_pool2 | 2.4.2 | 2.4.2 | https://commons.apache.org/proper/commons-pool/download_pool.cgi |
+| aaf_shib_ext | 1.0.1 | 1.1.1 | AAF |
```
Comment (rianniello) [inline README_libraries.md]: https://github.com/ausaccessfed/aaf-shib-ext is the link

**Comment by rianniello** [comment]
Would it make more sense just to overwrite the dependencies without having the manually manage them all? 

Adding a old version to the list could be error prone / easy to forget:
https://github.com/ausaccessfed/shibboleth-idp-installer/pull/198/commits/b65b8c5432b08b6e8caa8c0a054f2f9b736904dc

We solved this problem in hosted-idp-ansible... see
https://github.com/ausaccessfed/hosted-idp-ansible/blob/8ef151e1757fbf34fcc45ddbbbf3c6f58c3255b5/roles/idp/tasks/idp-install.yml#L56-L56
then
https://github.com/ausaccessfed/hosted-idp-ansible/blob/313269ce25035537af553688025b12b5e21a3a28/main.yml#L80-L80

Basically the jar ends up being stored on the IdP box as `aaf-shib-ext.jar`, and it's ansible's responsibility to push / overwrite the version we have specified in the config.

File: README_libraries.md (hunk 1 above)
Comment (trsau) [inline README_libraries.md]: I'm fine with just AAF as the link is to a private repo and this is a public document.

Comment (rianniello) [inline README_libraries.md]: No problems, I thought we had made that repo public.

**Comment by trsau** [comment]
> Would it make more sense just to overwrite the dependencies without having the manually manage them all?

Not sure what you mean by this?


> Adding a old version to the list could be error prone / easy to forget:
ausaccessfed/shibboleth-idp-installer@b65b8c5

This is the complete list ever. It captures all old versions of libraries that may have been deployed up to this point. With the new method of managing jars there will never be any reason to add to the list.

> We solved this problem in hosted-idp-ansible... see
https://github.com/ausaccessfed/hosted-idp-ansible/blob/8ef151e1757fbf34fcc45ddbbbf3c6f58c3255b5/roles/idp/tasks/idp-install.yml#L56-L56
> then
> https://github.com/ausaccessfed/hosted-idp-ansible/blob/313269ce25035537af553688025b12b5e21a3a28/main.yml#L80-L80

> Basically the jar ends up being stored on the IdP box as aaf-shib-ext.jar, and it's ansible's responsibility to push / overwrite the version we have specified in the config.

Yes, I did discuss this with Shaun. The reason I went with a symbolic link and let ansible maintain the link rather than the file was allow the IdP admin to determine the actual version of the installed .jar. Storing on the IdP box as add-shib-ext.jar we lose the version number. Getting IdP admins to workout what version is installed will be more difficult.

**Comment by rianniello** [comment]
>>Would it make more sense just to overwrite the dependencies without having the manually manage them all?
>Not sure what you mean by this?

I just mean a versionless jar file for the aaf-shib-ext (and those other deps). The content of it will replaced by ansible (with whatever version is configured).

> Yes, I did discuss this with Shaun. The reason I went with a symbolic link and let ansible maintain the link rather than the file was allow the IdP admin to determine the actual version of the installed .jar. Storing on the IdP box as add-shib-ext.jar we lose the version number. Getting IdP admins to workout what version is installed will be more difficult.

Ok this make sense to me, this information will be important. They technically could pull up the version from the local shibboleth-idp-installer git repo but this method may be more obvious / helpful to users.
